core/src/main/java/org/elasticsearch/action/index/TransportIndexAction.java

```diff
@@ -88,6 +90,10 @@ protected void doExecute(final IndexRequest request, final ActionListener<IndexR
         // if we don't have a master, we don't have metadata, that's fine, let it find a master using create index API
         ClusterState state = clusterService.state();
         if (autoCreateIndex.shouldAutoCreate(request.index(), state)) {
+            if (!settings.getAsBoolean(MapperService.INDEX_MAPPER_DYNAMIC_SETTING, MapperService.INDEX_MAPPER_DYNAMIC_DEFAULT)) {
+                String message = String.format("trying to auto create mapping for [%s] in index [%s], but dynamic mapping is disabled", request.type(), request.index());
+                throw new TypeMissingException(null, request.type(), message);
+            }
```

Comment:
Can you fold this logic into AutoCreateIndex so that it will also apply to delete/bulk/update requests?